src/app/clusters/localization-configuration-server/localization-configuration-server.cpp

```diff
@@ -123,21 +92,98 @@ CHIP_ERROR LocalizationConfigurationAttrAccess::Read(const ConcreteReadAttribute
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR LocalizationConfigurationAttrAccess::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+bool isSupportedLocales(chip::CharSpan newLangtag, chip::CharSpan & validLangtag)
 {
-    VerifyOrDie(aPath.mClusterId == LocalizationConfiguration::Id);
+    DeviceLayer::AttributeList<chip::CharSpan, DeviceLayer::kMaxLanguageTags> supportedLocales;
 
-    switch (aPath.mAttributeId)
+    if (DeviceLayer::PlatformMgr().GetSupportedLocales(supportedLocales) == CHIP_NO_ERROR)
+    {
+        for (auto locale : supportedLocales)
+        {
+            if (locale.data_equal(newLangtag))
+            {
+                return true;
+            }
+
+            validLangtag = locale;
```

Comment:
So this is pointing into the `supportedLocales` stack temporary, right?  When the stack unwinds, this will be pointing to garbage memory.  It looks to me like we need to have the consumer provide the memory for the outparam and copy our final supported locale into it....

Reply:
I don't think the helper function make the logic simpler here. Just use previous solution.